src/jdk.compiler/share/classes/com/sun/tools/javac/comp/Attr.java

```diff
@@ -2587,17 +2587,22 @@ public void visitApply(JCMethodInvocation tree) {
                         chk.checkRefType(qualifier.pos(),
                                          attribExpr(qualifier, localEnv,
                                                     encl));
-                    } else if (methName == names._super) {
-                        // qualifier omitted; check for existence
-                        // of an appropriate implicit qualifier.
-                        checkNewInnerClass(tree.meth.pos(), localEnv, site, true);
                     }
                 } else if (tree.meth.hasTag(SELECT)) {
                     log.error(tree.meth.pos(),
                               Errors.IllegalQualNotIcls(site.tsym));
                     attribExpr(((JCFieldAccess) tree.meth).selected, localEnv, site);
                 }
 
+                if (tree.meth.hasTag(IDENT)) {
```

Comment:
I've moved this code outside the `if (encl != null)` so that now it can be applied to all kind of superclasses. Note that `checkNewInnerClass` will also check whether the target class is either inner or local, so we don't need to check anything else here.